File: Week -3/Qs6.py
def reverse_words_in_string(s):
    # Split the string into words
    words = s.split()
    
    # Reverse each word and collect them in a list
    reversed_words = [word[::-1] for word in words]
    
    # Join the reversed words back into a single string
    reversed_string = ' '.join(reversed_words)
    
    return reversed_string

# Example usage
input_string = "C Programming Class"
output_string = reverse_words_in_string(input_string)
print(f"Original string: {input_string}")
print(f"Reversed words string: {output_string}")

# OUTPUT
# Original string: C Programming Class
# Reversed words string: C gnimmargorP ssalC

File: Week -3/Qs6.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

void reverseWord(char* start, char* end) {
    char temp;
    while (start < end) {
        temp = *start;
        *start = *end;
        *end = temp;
        start++;
        end--;
    }
}

void reverseWordsInString(char* str) {
    char* word_start = NULL;
    char* temp = str; // Temp pointer to traverse the string

    // Reverse individual words in the string
    while (*temp) {
        if ((word_start == NULL) && !isspace(*temp)) {
            word_start = temp;
        }
        if (word_start && (isspace(*temp) || *(temp + 1) == '\0')) {
            reverseWord(word_start, (isspace(*temp) ? temp - 1 : temp));
            word_start = NULL;
        }
        temp++;
    }
}

int main() {
    char str[] = "C Programming Class";
    printf("Original string: %s\n", str);
    
    reverseWordsInString(str);
    
    printf("Reversed words string: %s\n", str);
    
    return 0;
}

// OUTPUT
// Original string: C Programming Class
// Reversed words string: C gnimmargorP ssalC